tools/docs/generate_api_rst.py

```diff
@@ -0,0 +1,33 @@
+import os
+import shutil
+import sys
+import tarfile
+
+
+def main():
+    proto_srcs = sys.argv[1]
+    envoy_api_rst_files = sys.argv[1:-1]
+    output_filename = sys.argv[-1]
+
+    with open(proto_srcs) as f:
+        envoy_api_protos = [
+            f"{src.split('//')[1].replace(':', '/')}.rst" for src in f.read().split("\n") if src
```

Comment:
This seems magical; can you give before/after of an example transform and why this is done.